services/datamanager/builtin/builtin.go

```diff
@@ -679,3 +712,19 @@ func (svc *builtIn) updateDataCaptureConfigs(
 func generateMetadataKey(component, method string) string {
 	return fmt.Sprintf("%s/%s", component, method)
 }
+
+//nolint:unparam
+func pollFilesystem(ctx context.Context, wg *sync.WaitGroup, captureDir string, syncer *datasync.Manager, logger logging.Logger) {
+	for {
+		t := time.NewTimer(filesystemPollInterval)
```

Comment:
If yall have any questions about this lmk, but basically this allows us to create a timer thatll run on an interval AND be cancelled as soon as the context is. If we just used timer.sleep() we would cancel the context and have to wait for the sleep to finish, which could leave to viam-server taking up to 30 seconds to shutdown which isn't ideal. This  allows us to cancel the context and immediately exit gracefully without any leaking go threads.

Reply:
For my own understanding since it seems you've compared different functions, how does this compare to use a [clock.Ticker](https://github.com/viamrobotics/rdk/blob/0346bf027274d7606dd2c8121b6be2d49b46e65f/services/datamanager/builtin/builtin.go#L560-L577)?

Reply:
Ah, I didn't see the ticker implementation. I'm going to switch to that. After looking online, ticker will fire repeatedly on an interval vs time.NewTimer fires once, which is why I recreate it in each iteration. Thank you for pointing out!